install/installer/pkg/components/proxy/service.go

```diff
@@ -14,11 +14,29 @@ import (
 	"k8s.io/apimachinery/pkg/runtime"
 )
 
+var allowedServiceTypes = map[corev1.ServiceType]struct{}{
+	corev1.ServiceTypeLoadBalancer: {},
+	corev1.ServiceTypeClusterIP:    {},
+	corev1.ServiceTypeNodePort:     {},
+	corev1.ServiceTypeExternalName: {},
+}
+
 func service(ctx *common.RenderContext) ([]runtime.Object, error) {
+	serviceType := corev1.ServiceTypeLoadBalancer
+
 	loadBalancerIP := ""
 	_ = ctx.WithExperimental(func(cfg *experimental.Config) error {
-		if cfg.WebApp != nil && cfg.WebApp.ProxyConfig != nil && cfg.WebApp.ProxyConfig.StaticIP != "" {
-			loadBalancerIP = cfg.WebApp.ProxyConfig.StaticIP
+		if cfg.WebApp != nil && cfg.WebApp.ProxyConfig != nil {
+			if cfg.WebApp.ProxyConfig.StaticIP != "" {
+				loadBalancerIP = cfg.WebApp.ProxyConfig.StaticIP
```

Comment:
Do we need to tie this to the `ServiceType`? :thinking: Really not sure what happens if we set this field, but do not have `LoadBalancer` type.

Reply:
You mean the bit here right? https://github.com/gitpod-io/gitpod/pull/10537/files#diff-d9a72ab10ad2fef022eeb810eac2018092ca6dec333c9c38c5c5c8c5244bafc0R79
It doesn't misbehave if we leave the `loadbalancerIP` in, I just didn't want to make the service definition confusing. I can roll this back.

Reply:
:heavy_check_mark:

Reply:
Scratch my comment, now I get what you mean. https://github.com/gitpod-io/gitpod/pull/10537/files#diff-d9a72ab10ad2fef022eeb810eac2018092ca6dec333c9c38c5c5c8c5244bafc0R79 this saves from setting the `loadbalancerIP` if the `ServiceType` is not loadbalancer. This reduces any potential confusion, although shouldn't effect the service in any ways.

I put  the check logic here just because the `webapp` experimental section was being parsed anyway, wanted to make the bust of the if jump.

Reply:
>  this saves from setting the loadbalancerIP if the ServiceType is not loadbalancer.

Ah, thanks, missed that! :bow: